code_doc_gen/ai_providers/groq.py

```
@@ -0,0 +1,57 @@
+from typing import Optional, List, Union
+from .base import AIProvider
+
+try:
+    import groq
+    GROQ_AVAILABLE = True
+except ImportError:
+    GROQ_AVAILABLE = False
+
+class GroqProvider(AIProvider):
+    def __init__(self, api_key: str, model: Union[str, List[str]] = None):
```

Ruff (0.14.13)

PEP 484 prohibits implicit `Optional`

Convert to `T | None`

(RUF013)